core-blocks/latest-posts/edit.js

```diff
@@ -146,8 +148,8 @@ class LatestPostsEdit extends Component {
 						<li key={ i }>
 							<a href={ post.link } target="_blank">{ decodeEntities( post.title.rendered.trim() ) || __( '(Untitled)' ) }</a>
 							{ displayPostDate && post.date_gmt &&
-								<time dateTime={ moment( post.date_gmt ).utc().format() } className={ `${ this.props.className }__post-date` }>
-									{ moment( post.date_gmt ).local().format( 'MMMM DD, Y' ) }
+								<time dateTime={ format( 'c', post.date_gmt ) } className={ `${ this.props.className }__post-date` }>
+									{ dateI18n( dateFormat, post.date_gmt ) }
```

Comment:
It's nice that this avoids using `moment` explicitly. I'd have preferred that the `date` module uses the configured format automatically if no format is provided, this probably requires some refactoring to its API, so worth exploring separately.